### PR TITLE
DAOS-9198: Clear up daos_pool_connect*().

### PR DESCRIPTION
**Pull request by wiliamhuang**
Only daos_pool_connect2 is provided in API. Pool uuid_t support is dropped.

We only provide daos_pool_connect2() in our API. daos_pool_connect() should be
used in other places. pool uuid string is NEEDED as an argument. uuid_t is not
supported any more.

### CODE REVIEW

**Comment by wangshilong** [comment]
A lot of failure because of following errors:

/lib64/libdaos.so.2: undefined symbol: daos_pool_connect

**Comment by wiliamhuang** [comment]
> A lot of failure because of following errors:
> 
> /lib64/libdaos.so.2: undefined symbol: daos_pool_connect

I just replaced daos_pool_connect by daos_pool_connect2 in src/client/pydaos/raw/daos_api.py. The issue should be resolved.

File: src/include/daos_pool.h [hunk 1]
```diff
@@ -190,7 +190,7 @@ struct daos_pool_cont_info {
  *			-DER_NONEXIST	Pool is nonexistent
  */
 int
-daos_pool_connect(const char *pool, const char *sys, unsigned int flags,
+daos_pool_connect2(const char *pool, const char *sys, unsigned int flags,
 		  daos_handle_t *poh, daos_pool_info_t *info, daos_event_t *ev);
```
Comment (johannlombardi) [inline src/include/daos_pool.h:194]: I think that we want to keep advertising the daos_pool_connect symbol here and redirect it to daos_pool_connect2 with a #define.

File: src/tests/suite/daos_test_common.c [hunk 2]
```diff
@@ -628,8 +628,8 @@ d_rank_t ranks_to_kill[MAX_KILLS];
 bool
 test_runable(test_arg_t *arg, unsigned int required_nodes)
 {
-	int		 i;
-	static bool	 runable = true;
+	int	i;
+	int	runable = 1;
```
Comment (johannlombardi) [inline src/tests/suite/daos_test_common.c:632]: was landed in a different patch IIRC and not relevant to this one.

File: src/client/api/pool.c [hunk 3]
```diff
@@ -14,36 +14,6 @@
...
-	if (rc)
-		return rc;
-
-	args = dc_task_get_args(task);
-	args->grp	= grp;
-	args->flags	= flags;
-	args->poh	= poh;
-	args->info	= info;
-	/** use deprecated field for backward compatibility */
-	uuid_copy((unsigned char *)args->uuid, uuid);
```
Comment (johannlombardi) [inline src/client/api/pool.c:41]: args->uuid can also likely be removed from daos_pool_connect_t, but this can be cleaned up in a separate patch.

**Review by johannlombardi — CHANGES_REQUESTED** [review]
Just a few minor changes. Issues with the CI should also be addressed. Have you merged master?

File: src/client/api/pool.c (hunk 3 above)
Comment (wiliamhuang) [reply src/client/api/pool.c:41]: ok. I can remove "uuid_t	uuid" from daos_pool_connect_t in a separate patch.

File: src/include/daos_pool.h (hunk 1 above)
Comment (wiliamhuang) [reply src/include/daos_pool.h:194]: We do have "#define daos_pool_connect daos_pool_connect2" at the end of src/include/daos_pool.h (this file). We still use "daos_pool_connect" in other c code.

File: src/tests/suite/daos_test_common.c (hunk 2 above)
Comment (wiliamhuang) [reply src/tests/suite/daos_test_common.c:632]: Right. I accidentally included src/tests/suite/daos_test_common.c in one commit of this PR. I intended to use the current latest daos_test_common.c to replace the one I modified. I need to revert my change of daos_test_common.c which is a part of my first commit in this PR. Shall I use "git revert"? Thank you!

File: src/include/daos_pool.h (hunk 1 above)
Comment (wangdi1) [reply src/include/daos_pool.h:194]: Hmm, a bit confused here, if daos_pool_connect2 will not be exported anyway. why not remove daos_pool_connect2 and  just use daos_pool_connect() here?

Comment (wiliamhuang) [reply src/include/daos_pool.h]: I just re-arranged the macro of daos_pool_connect. Please check out commit 5. Thank you!

File: src/include/daos_pool.h [hunk 4]
```diff
@@ -9,6 +9,8 @@
 #ifndef __DAOS_POOL_H__
 #define __DAOS_POOL_H__
 
+#define daos_pool_connect daos_pool_connect2
```
Comment (wangdi1) [inline src/include/daos_pool.h:12]: Hmm, do we still need this?

File: src/client/pydaos/raw/daos_api.py [hunk 5]
```diff
@@ -2282,7 +2287,7 @@ def __init__(self, path):
             'close-obj':       self.libdaos.daos_obj_close,
             'close-tx':        self.libdaos.daos_tx_close,
             'commit-tx':       self.libdaos.daos_tx_commit,
-            'connect-pool':    self.libdaos.daos_pool_connect,
+            'connect-pool':    self.libdaos.daos_pool_connect2,
```
Comment (wangdi1) [inline src/client/pydaos/raw/daos_api.py:2290]: daos_pool_connect?

File: src/include/daos_pool.h (hunk 4 above)
Comment (wiliamhuang) [reply src/include/daos_pool.h:12]: Sorry. Just found your comment. Johann suggested to keep this macro to use daos_pool_connect2() in the API provided by libdaos.so. In C/C++, we use daos_pool_connect(). We have to use daos_pool_connect2() since it is the only function name exposed.

File: src/client/pydaos/raw/daos_api.py (hunk 5 above)
Comment (wiliamhuang) [reply src/client/pydaos/raw/daos_api.py:2290]: Right. Johann suggested to exposed daos_pool_connect2() only. Other languages need to call daos_pool_connect2.

**Comment by wiliamhuang** [comment]
> Just a few minor changes. Issues with the CI should also be addressed. Have you merged master?
With merging latest master, CI issues seem resolved.

**Review by wangshilong — APPROVED** [review]

File: src/include/daos_pool.h (hunk 1 above)
Comment (wiliamhuang) [reply src/include/daos_pool.h]: After merging master branch, CI issue is gone.
